### PR TITLE
feat(sounds): add SoundManager class

### PR DESCRIPTION
**Pull request by molimauro**
Jira ref: SA-43

### CODE REVIEW

File: plugins/local/classLoader.ts [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-import WebRTC from '~/utilities/WebRTC/WebRTC'
 import SolanaManager from '~/utilities/SolanaManager/SolanaManager'
+import SoundManager from '~/utilities/SoundManager/SoundManager'
+import WebRTC from '~/utilities/WebRTC/WebRTC'
 
 declare module 'vue/types/vue' {
   interface Vue {
     $WebRTC: typeof WebRTC
     $SolanaManager: typeof SolanaManager
+    $sounds: SoundManager
```
Comment (RetroPronghorn) [inline plugins/local/classLoader.ts:12]: Let's keep classes title case so simply change to `$Sounds`

File: plugins/local/classLoader.ts [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
   }
 }
 
 declare module '@nuxt/types' {
   interface Context {
     $WebRTC: typeof WebRTC
     $SolanaManager: typeof SolanaManager
+    $sounds: SoundManager
```
Comment (RetroPronghorn) [inline plugins/local/classLoader.ts:20]: Same thing here, `$Sounds`

File: utilities/SoundManager/SoundManager.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { Howl } from 'howler'
+import { Config } from '~/config'
+
+// Keep this type in sync with Config.sounds
+export enum Sounds {
+  NewMessage = 'newMessage',
```
Comment (RetroPronghorn) [inline utilities/SoundManager/SoundManager.ts:6]: Let's make enums all uppercase, I don't know if it's a standard but it's what I've been doing for the other enums so it's clear they are enums without digging in. so `NEW_MESSAGE` `CALL` `HANGUP`.. etc.

File: utilities/SoundManager/SoundManager.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+   * @constructs SoundManager
+   */
+  constructor() {
+    this.sounds = {} as Record<Sounds, Howl>
+    for (const [key, value] of Object.entries(Config.sounds) as [
+      Sounds,
+      string
+    ][]) {
+      this.sounds[key] = new Howl({
+        src: [`${Config.ipfs.browser}${value}`],
```
Comment (RetroPronghorn) [inline utilities/SoundManager/SoundManager.ts:33]: let's change this in the config and here to `Config.ips.gateway` this was a terminology mistake I made early on.

File: utilities/SoundManager/SoundManager.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  /** @function
+   * Plays a specified sound
+   * @name play
+   * @argument sound Name of the sound file to play.
+   * @returns null
+   */
+  playSound(sound: Sounds) {
+    if (!this.sounds[sound]) {
```
Comment (RetroPronghorn) [inline utilities/SoundManager/SoundManager.ts:48]: We use this logic a few times, is there a way we can make a private method to check if the sound exists and if not throw the error and stop execution?

File: components/tailored/core/media/actions/Actions.vue [hunk 6]
```diff
@@ -3,11 +3,21 @@
...
 
 export default Vue.extend({
   computed: {
     ...mapState(['audio']),
   },
+  methods: {
+    toggleMute() {
+      const muted = this.$store.state.audio.muted
+      if (!muted) this.$sounds.playSound(Sounds.Mute)
+      if (muted) this.$sounds.playSound(Sounds.Unmute)
```
Comment (RetroPronghorn) [inline components/tailored/core/media/actions/Actions.vue:16]: let's use else logic here

**Review by RetroPronghorn — CHANGES_REQUESTED** [review]

**Comment by molimauro** [comment]
PR updated! Let me know if this is ok ;)

**Review by RetroPronghorn — APPROVED** [review]

**Pull request merged**